### DISCUSSION TITLE
CKB Dev Log 2023-10-04

### DISCUSSION THREAD

**Discussion by anipenguin** [nervosnetwork/ckb#4174] [Dev Log]
![image](https://github.com/nervosnetwork/ckb/assets/112425356/76399ef4-3102-4f14-8210-cbddd676a3f5)
📸 by @chenyukang 

### ***Features***

- Introduced new and enhanced RPCs for CKB2023 in lumos
    - https://github.com/ckb-js/lumos/pull/556
    - https://github.com/ckb-js/lumos/pull/558

### ***Improvements***

- Completed end-to-end testing for CKB-Auth and provided documentation for Bitcoin and Ethereum
    - https://github.com/nervosnetwork/ckb-auth/pull/10
    - https://github.com/nervosnetwork/ckb-auth/pull/12

### ***Fixes***

- Rectified the serialization issue with the **`data2`** parameter in lumos
    - https://github.com/ckb-js/lumos/pull/555

## In Pipeline…

- Addressing memory limitations in CKB-VM
    - WIP: https://github.com/nervosnetwork/ckb-vm/pull/378
- Exploring two approaches to enhance [[Cells Commitments](https://github.com/nervosnetwork/rfcs/pull/424#)](https://github.com/nervosnetwork/rfcs/pull/424#)
    - Streamlining historical record storage methods
    - Optimizing Merkle Mountain Range (MMR) storage read and write processes
    - WIP: https://github.com/nervosnetwork/sparse-merkle-tree/pull/47
- Adding troubleshoot and metric API for CKB’s transaction pool
    - WIP: https://github.com/nervosnetwork/ckb/pull/4173
- Migrating from jsonrpc to jsonrpc-utils as the RPC framework to enable support for more upcoming features
    - WIP: https://github.com/nervosnetwork/ckb/pull/4170

**Comment by joshyates1980** [comment]
Is the --migration parameter required for this version?  Or can I copy over the Data folder from previous ckb version?

**Reply by roughpandaz** [reply]
@joshyates1980 good quesion. @chenyukang is this needed?

**Reply by chenyukang** [reply]
There were no break changes this week, it's Ok to copy over the data folder.
